crates/sui-rpc-benchmark/src/json_rpc/runner.rs

```diff
@@ -1,38 +1,59 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use super::request_loader::JsonRpcRequestLine;
+use crate::config::BenchmarkConfig;
 /// This module implements the JSON RPC benchmark runner.
 /// The main function is `run_queries`, which runs the queries concurrently
 /// and records the overall and per-method stats.
-use anyhow::Result;
+use anyhow::{bail, Context as _, Result};
+use dashmap::DashMap;
+use phf::phf_map;
+use serde_json::Value;
 use std::{
     collections::HashMap,
     sync::{Arc, Mutex},
-    time::Instant,
+    time::{Duration, Instant},
 };
 use sui_indexer_alt_framework::task::TrySpawnStreamExt;
 use tokio::time::timeout;
+use tracing::info;
 
-use super::request_loader::JsonRpcRequestLine;
-use crate::config::BenchmarkConfig;
+/// static map of method names to the index of their cursor parameter
+static METHOD_CURSOR_POSITIONS: phf::Map<&'static str, usize> = phf_map! {
+    // based on function headers in crates/sui-json-rpc-api/src/indexer.rs
+    "suix_getOwnedObjects" => 2,
+    "suix_queryTransactionBlocks" => 1,
+    // based on function headers in crates/sui-json-rpc-api/src/coin.rs
+    "suix_getCoins" => 2,
+    "suix_getAllCoins" => 1,
+};
 
+/// Statistics for a single JSON RPC method
 #[derive(Clone, Default)]
 pub struct PerMethodStats {
     pub total_sent: usize,
     pub total_errors: usize,
-    // record total latency and calculate average latency later to avoid duplicate calculations
     pub total_latency_ms: f64,
 }
 
+/// Aggregated statistics for all JSON RPC requests
```

Comment:
technically `JsonRpcStats` struct can be re-written with `DashMap`, `AtomicUsize` etc. but it's even more verbose and not more reliable / efficient then the existing `lock()`, so I decided to leave this as it is.